packages/regl-worldview/src/types/index.js

```diff
@@ -194,6 +184,23 @@ export type PolygonType = BaseShape & {
   points: (Point | Vec3)[],
 };
 
+export type MouseEventObject = {
+  object: ?BaseShape,
+  instanceIndex: ?number,
+};
+
+export type HitmapId = number;
+/*
+ * type: "instanced" if mapping multiple IDs to a single callback object, "single" if 1:1.
+ * callbackObject: the object to pass to event callbacks when this object is interacted with.
+ * count: If "instanced", how many IDs to map to the callback object.
+ * return type: an array of the IDs assigned.
+ */
+export type AssignNextIdsFn = (
+  { type: "single", callbackObject: BaseShape } | { type: "instanced", callbackObject: BaseShape, count: number }
```

Comment:
📜

- Strange to have "callback" in the name; why not just call it `object` since that is the terminology we use elsewhere?

- Is there any need for the object to be BaseShape (doesn't seem like it's really used inside the HitmapIdManager)? Are Commands allowed to have children that don't derive from BaseShape?

- Maybe make this into a new type so you can use it again in HitmapIdManager.js

- Is there actually anything related to instancing here, or should we just rename it `"multiple"`?

  - Do we really need a `type` or could we instead just rely on `count === 1`?


- What if we also/instead return colors from this function? Consumers are pretty much always going to be converting this into a color with intToRGB, right?

Reply:
* We could just call it object
* We currently don't support it in our typing system because of these interaction changes, but I could change that
* This type is almost identical to that type, but can't be used exactly because it's wrapped. Sharing a type signature would be a huge pain in Flow
* Kind of? The main reason you'd assign multiple IDs to the same object is for instancing. I chose to make it explicit based on the name but I could change it to be based on the count.
* We could also return colors with this function